sdk/core/Azure.Core/src/Shared/IBearerTokenChallengeOptions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Options to be exposed in client options classes related to bearer token authorization challenge scenarios.
+    /// </summary>
+    internal interface IBearerTokenChallengeOptions
+    {
+        /// <summary>
+        ///  Disables tenant discovery through the authorization challenge when the client is configured to use a <see cref="TokenCredential"/>.
+        /// When disabled, the client will not attempt an initial un-authroized request to prompt a challenge in order to discover the correct tenant for the resource.
+        /// </summary>
+        public bool DisableTenantDiscovery { get; }
```

Comment:
When would I, as a customer, want to disable the TenantDiscovery?

Reply:
If, for some reason, it's important to you not to every issue an initial unauthorized request to prompt a challenge.

Reply:
Can you give me an example from one of our services?

Reply:
I'm not totally sold that this is important, but because this changes the behavior of how a client authorizes to the service, I thought it may be important to allow an option to maintain the current behavior which is to attempt to authorize the request on the first request rather than send an unauthorized request, receive a challenge, then authorize the request.

Reply:
Seems that this is something that client authors should be in control of but maybe not the end customers.